src/rpcx/client.py

```diff
@@ -154,8 +160,12 @@ def _make_req_task(self, req_id: int) -> Generator[_RequestTask, None, None]:
             del self.tasks[req_id]
 
     async def request(self, method: str, *args: Any, **kwargs: Any) -> Any:
+        if self._closed:
+            raise anyio.ClosedResourceError(f"Cannot request {method}, server stream is closed")
```

Comment:
I'm not sure it's correct to raise this kind of exception here.  The class doesn't inherit an anyio stream or anything like that so it's a little surprising to raise this kind of exception.  

My gut would be to raise a different builtin exception but im open to debate.